system-tests/e2e-transfer-test/runner/src/test/java/org/eclipse/dataspaceconnector/test/e2e/AbstractEndToEndTransfer.java

```diff
@@ -50,29 +52,63 @@ void httpPullDataTransfer() {
 
         assertThat(contractAgreementId).isNotEmpty();
 
-        var transferProcessId = CONSUMER.dataRequest(contractAgreementId, assetId, PROVIDER, sync());
+        var dataRequestId = UUID.randomUUID().toString();
+        var transferProcessId = CONSUMER.dataRequest(dataRequestId, contractAgreementId, assetId, PROVIDER, sync());
 
         await().atMost(timeout).untilAsserted(() -> {
             var state = CONSUMER.getTransferProcessState(transferProcessId);
             assertThat(state).isEqualTo(COMPLETED.name());
         });
 
-        await().atMost(timeout).untilAsserted(() -> {
-            given()
-                    .baseUri(CONSUMER.backendService().toString())
-                    .when()
-                    .get("/api/consumer/data")
-                    .then()
-                    .statusCode(200)
-                    .body("message", equalTo("some information"));
-        });
+        // retrieve the data reference
+        await().atMost(timeout).untilAsserted(() -> given()
+                .baseUri(CONSUMER.backendService().toString())
+                .when()
+                .get("/api/consumer/dataReference/" + dataRequestId)
+                .then()
+                .statusCode(200)
+                .extract()
+                .body()
+                .as(EndpointDataReference.class));
+
+        var edr = given()
+                .baseUri(CONSUMER.backendService().toString())
+                .when()
+                .get("/api/consumer/dataReference/" + dataRequestId)
+                .then()
+                .statusCode(200)
+                .extract()
+                .body()
+                .as(EndpointDataReference.class);
+
+        // pull the data
+        await().atMost(timeout).untilAsserted(() -> given()
```

Comment:
Maybe also this part could be included in the same method I proposed above: the method on participant could be like `getData` or `pullData` and it could hide all the edr details

Reply:
Updated, could you please have another look?